src/planning.ts

```diff
@@ -476,109 +520,115 @@ function dedupPoints(points: Vec2[], epsilon: number): Vec2[] {
         new Block(-accel, z.t3, vMax, z.p3, z.p4)
-      ];
-      nextSegment.entryVelocity = vExit;
-      i += 1;
+      ]
+      nextSegment.entryVelocity = vExit
+      i += 1
     } else {
       // Accelerate, then decelerate.
       segment.blocks = [
         new Block(accel, m.t1, vInitial, m.p1, m.p2),
         new Block(-accel, m.t2, m.vMax, m.p2, m.p3)
-      ];
-      nextSegment.entryVelocity = vExit;
-      i += 1;
+      ]
+      nextSegment.entryVelocity = vExit
+      i += 1
     }
   }
-  const blocks: Block[] = [];
+  const blocks: Block[] = []
   segments.forEach((s) => {
     s.blocks.forEach((b) => {
       if (b.duration > epsilon) {
-        blocks.push(b);
+        blocks.push(b)
       }
-    });
-  });
-  return new XYMotion(blocks);
+    })
+  })
+  return new XYMotion(blocks)
 }
 
-export function plan(
+export function plan (
   paths: Vec2[][],
   profile: ToolingProfile
 ): Plan {
-  const motions: Motion[] = [];
-  let curPos = { x: 0, y: 0 };
-  const penMaxUpPos = profile.penUpPos < profile.penDownPos ? 100 : 0
-  // for each path: move to the initial point, put the pen down, draw the path,
-  // then pick the pen up.
-  paths.forEach((p, i) => {
-    const m = constantAccelerationPlan(p, profile.penDownProfile);
-    const penUpPos = i === paths.length - 1 ? Device.Axidraw.penPctToPos(penMaxUpPos) : profile.penUpPos;
-    motions.push(
-      constantAccelerationPlan([curPos, m.p1], profile.penUpProfile),
-      new PenMotion(profile.penUpPos, profile.penDownPos, profile.penDropDuration),
-      m,
-      new PenMotion(profile.penDownPos, penUpPos, profile.penLiftDuration)
-    );
-    curPos = m.p2;
-  });
-  // finally, move back to (0, 0).
-  motions.push(constantAccelerationPlan([curPos, {x: 0, y: 0}], profile.penUpProfile));
-  motions.push(new PenMotion(Device.Axidraw.penPctToPos(penMaxUpPos), profile.penUpPos, profile.penDropDuration));
-  return new Plan(motions);
+  const motions: Motion[] = []
+  let curPos = { x: 0, y: 0 }
+
+  const penMotions = {
+    up: new PenMotion(profile.penDownPos, profile.penUpPos, profile.penLiftDuration),
+    down: new PenMotion(profile.penUpPos, profile.penDownPos, profile.penDropDuration)
+  }
+
+  // For each path - move to the initial position, put the pen down, draw the path, bring pen up
+  paths.forEach(path => {
+    const motion = constantAccelerationPlan(path, profile.penDownProfile)
+    const position = constantAccelerationPlan([curPos, motion.p1], profile.penUpProfile)
+
+    motions.push(position, penMotions.down, motion, penMotions.up)
+    curPos = motion.p2
+  })
+
+  // Move to {x: 0, y: 0}
+  motions.push(constantAccelerationPlan([curPos, { x: 0, y: 0 }], profile.penUpProfile))
+  // TODO: should minPenPosition be
+  //      a) profile.penUpPosition?
+  //      b) profile.penDownPosition?
+  //      c) device.penPctToPos(0)
+  //      d) device.penPctToPos(100)
+  //      e) device.penPctToPos(profile.penUpPos < profile.penDownPos ? 100: 0)
```

Comment:
Can remove this, it looks like it goes back to pen up position correctly on brushless, but someone should check on regular.

Reply:
I think I need to write up an issue that captures all the penUp/penDown positioning oddities.